Source/items.cpp

```diff
@@ -4204,14 +4226,26 @@ void CreateSpellBook(int x, int y, int ispell, BOOL sendmsg, BOOL delta)
 	BOOL done;
 
 	done = FALSE;
+#ifdef HELLFIRE
+	int lvl = spelldata[ispell].sBookLvl + 1;
+	if (lvl < 1) {
+		return;
+	}
+	idx = RndTypeItems(0, IMISC_BOOK, lvl);
+#else
 	idx = RndTypeItems(0, IMISC_BOOK);
+#endif
 	if (numitems < MAXITEMS) {
 		ii = itemavail[0];
 		GetSuperItemSpace(x, y, ii);
 		itemavail[0] = itemavail[MAXITEMS - numitems - 1];
 		itemactive[numitems] = ii;
 		while (!done) {
+#ifdef HELLFIRE
+			SetupAllItems(ii, idx, GetRndSeed(), 2 * lvl, 1, 1, FALSE, delta);

```

Comment:
```suggestion
			SetupAllItems(ii, idx, GetRndSeed(), 2 * lvl, 1, TRUE, FALSE, delta);
```
(SetupAllItems should also be updated)